src/thorlabs_cube/driver/tcube/tsc.py

```diff
@@ -0,0 +1,407 @@
+import struct as st
+
+from thorlabs_cube.driver.base import _Cube
+from thorlabs_cube.driver.message import MGMSG, Message, MsgError
+
+_CHANNEL = 0x01
+_RESERVED = 0x00
+
+
+class Tsc(_Cube):
+    """TSC001 T-Cube Motor Controller class"""
+
+    def __init__(self, *args, **kwargs):
+        _Cube.__init__(self, *args, **kwargs)
+        self.status_report_counter = 0
+
+    async def handle_message(self, msg):
+        msg_id = msg.id
+        data = msg.data
+
+        if msg_id == MGMSG.HW_DISCONNECT:
+            raise MsgError("Error: Please disconnect the TDC001")
+        elif msg_id == MGMSG.HW_RESPONSE:
+            raise MsgError(
+                "Hardware error, please disconnect " "and reconnect the TDC001"
+            )
+        elif msg_id == MGMSG.HW_RICHRESPONSE:
+            (code,) = st.unpack("<H", data[2:4])
+            raise MsgError(
+                "Hardware error {}: {}".format(code, data[4:].decode(encoding="ascii"))
+            )
+        elif (
+            msg_id == MGMSG.MOT_MOVE_COMPLETED
+            or msg_id == MGMSG.MOT_MOVE_STOP
+            or msg_id == MGMSG.MOT_GET_STATUSUPDATE
+        ):
+            if self.status_report_counter == 25:
+                self.status_report_counter = 0
+                await self.send(Message(MGMSG.MOT_MOVE_COMPLETED))
+            else:
+                self.status_report_counter += 1
+            (
+                self.position,
+                self.encoder_count,
+                self.status_bits,
+                self.chan_identity_two,
+                _RESERVED,
+                _RESERVED,
+                _RESERVED,
+            ) = st.unpack(
+                "<LLLHLLL",
+                data[2:],
+            )
+
+    def module_identify(self):
+        return super().module_identify()
+
+    def hardware_start_update_messages(self, update_rate):
+        return super().hardware_start_update_messages(update_rate)
+
+    def hardware_stop_update_messages(self):
+        return super().hardware_stop_update_messages()
+
+    def hardware_request_information(self):
+        return (
+            super().hardware_request_information()
+        )  # FIXME->need to change serialization format but should work after that
+
+    def is_channel_enabled(self):
+        return super().is_channel_enabled()
+
+    def ping(self):
+        return super().ping()
+
+    async def set_channel_enable_state(self, activated):
+        return await super().set_channel_enable_state(activated)
+
+    async def get_channel_enable_state(self):
+        return await super().get_channel_enable_state()
+
+    # functions from base class definition end
```

Comment:
```suggestion
```
Methods inherited from the base class at not overridden should not need to be redefined.